nucypher/blockchain/eth/agents.py

```diff
@@ -480,13 +480,14 @@ def withdraw(self, staker_address: str, amount: int):
         return receipt
 
     @validate_checksum_address
-    def get_flags(self, staker_address: str) -> Tuple[bool, bool, bool]:
-        wind_down_flag, restake_flag, measure_work_flag = self.contract.functions.getFlags(staker_address).call()
-        return wind_down_flag, restake_flag, measure_work_flag
+    def get_flags(self, staker_address: str) -> Tuple[bool, bool, bool, bool]:
+        flags = self.contract.functions.getFlags(staker_address).call()
```

Comment:
Without fully understanding the ramifications of this, can we add a function to the contract for the retrieval of a specific flag?

Reply:
it's compromise, because adding separate functions will increase size of contract

Reply:
Can we instead do it only on the python side?

It also saves us from having to add another element of the tuple everywhere we call `get_flags` when subsequent flags get added eg. the change made to tests in this PR to add `snapshots` as another member of the returned tuple.

Reply:
they are already there, for each flag (except measure work), get_flags is representation for contract method
also adding flag is not usual thing, it will be rare